src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -0,0 +1,232 @@
+import * as path from 'path';
+import { EslintStrategy } from "./BaseEslintEngine";
+import {FileHandler} from '../util/FileHandler';
+import {Config} from '../util/Config';
+import {Controller} from '../../ioc.config';
+import { SfdxError, Logger } from '@salesforce/core';
+import { OutputProcessor } from '../pmd/OutputProcessor';
+
+/**
+ * Type mapping to tsconfig.json files
+ */
+type TSConfig = {
+	compilerOptions: {
+		allowJs: boolean;
+		outDir: string;
+		outFile: string;
+	};
+	include: string[];
+	exclude: string[];
+	files: string[];
+}
+
+const ES_PLUS_TS_CONFIG = {
+	"parser": "@typescript-eslint/parser",
+	"extends": [
+		"eslint:recommended",
+		"plugin:@typescript-eslint/recommended",
+		"plugin:@typescript-eslint/eslint-recommended"
+	],
+	"parserOptions": {
+		"sourceType": "module",
+		"ecmaVersion": 2018,
+	},
+	"plugins": [
+		"@typescript-eslint"
+	],
+	"ignorePatterns": [
+		"lib/**",
+		"node_modules/**"
+	],
+	"useEslintrc": false // TODO derive from existing eslintrc if found and desired
+
+};
+
+const TS_CONFIG = "tsconfig.json";
+
+export class TypescriptEslintStrategy implements EslintStrategy {
+	private static ENGINE_NAME = "eslint-typescript";
+	private static LANGUAGE = ["typescript"];
+
+	private initialized: boolean;
+	protected logger: Logger;
+	private fileHandler: FileHandler;
+	private config: Config;
+	private outputProcessor: OutputProcessor;
+
+	public async init(): Promise<void> {
+		if (this.initialized) {
+			return;
+		}
+		this.logger = await Logger.child(this.getName());
+		this.fileHandler = new FileHandler();
+		this.config = await Controller.getConfig();
+		this.outputProcessor = await OutputProcessor.create({});
+		this.initialized = true;
+	}
+
+	isEnabled(): boolean {
+		return this.config.isEngineEnabled(this.getName());
+	}
+
+	getName(): string {
+		return TypescriptEslintStrategy.ENGINE_NAME;
+	}
+
+	isRuleKeySupported(): boolean {
```

Comment:
New change to allow non @typescript-eslint rules to run for Typescript as well since this gives additional coverage apart from typescript-only rules.